app/src/main/java/de/danoeh/antennapod/fragment/preferences/SynchronizationPreferencesFragment.java

```diff
@@ -0,0 +1,294 @@
+
+import org.greenrobot.eventbus.EventBus;
+import org.greenrobot.eventbus.Subscribe;
+import org.greenrobot.eventbus.ThreadMode;
+
+import static de.danoeh.antennapod.core.sync.SyncService.SYNC_PROVIDER_CHOICE_GPODDER_NET;
+
+public class SynchronizationPreferencesFragment extends PreferenceFragmentCompat {
+    private static final String PREFERENCE_SYNCHRONIZATION_DESCRIPTION = "preference_synchronization_description";
+    private static final String PREFERENCE_LOGIN = "pref_synchronization_authenticate";
+    private static final String PREF_GPODNET_SETLOGIN_INFORMATION = "pref_gpodnet_setlogin_information";
+    private static final String PREFERENCE_SYNC = "pref_synchronization_sync";
+    private static final String PREFERENCE_FORCE_FULL_SYNC = "pref_synchronization_force_full_sync";
+    private static final String PREFERENCE_LOGOUT = "pref_synchronization_logout";
+    public static final String SYNC_PROVIDER_UNSET = "unset";
+
+    @Override
+    public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
+        addPreferencesFromResource(R.xml.preferences_synchronization);
+        setupScreen();
+        updateScreen();
+    }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        ((PreferenceActivity) getActivity()).getSupportActionBar().setTitle(R.string.synchronization);
+        updateScreen();
+        EventBus.getDefault().register(this);
+    }
+
+    @Override
+    public void onStop() {
+        super.onStop();
+        EventBus.getDefault().unregister(this);
+        ((PreferenceActivity) getActivity()).getSupportActionBar().setSubtitle("");
+    }
+
+    @Subscribe(threadMode = ThreadMode.MAIN, sticky = true)
+    public void syncStatusChanged(SyncServiceEvent event) {
+        updateScreen();
+        if (!GpodnetPreferences.loggedIn()) {
+            return;
+        }
+        if (event.getMessageResId() == R.string.sync_status_error
+                || event.getMessageResId() == R.string.sync_status_success) {
+            updateLastSyncReport(SyncService.isLastSyncSuccessful(getContext()),
+                    SyncService.getLastSyncAttempt(getContext()));
+        } else {
+            ((PreferenceActivity) getActivity()).getSupportActionBar().setSubtitle(event.getMessageResId());
+        }
+    }
+
+    private void setupScreen() {
+        final Activity activity = getActivity();
+
+        findPreference(PREFERENCE_LOGIN).setOnPreferenceClickListener(preference -> {
+            AlertDialog.Builder builder = new AlertDialog.Builder(getContext());
+            builder.setTitle(R.string.dialog_choose_sync_service_title);
+            String[] syncProviderDescriptions = {
+                    getString(R.string.gpodnet_description),
+                    getString(R.string.preference_synchronization_summary_nextcloud)
+            };
+            int[] syncProviderIcons = {R.drawable.gpodder_icon, R.drawable.nextcloud_logo_svg};
```

Comment:
Shouldn't this also be in `SynchronizationProviderViewData`?

Reply:
good catch. i reused the enum structure